### PR TITLE
Add extraConfig for ssh module

### PR DESCRIPTION
**Pull request by luigy**

### CODE REVIEW

File: modules/programs/ssh/default.nix [hunk 1]
```diff
@@ -81,11 +90,14 @@ in
...
-        concatStringsSep "," h.hostNames + " "
-        + (if h.publicKey != null then h.publicKey else readFile h.publicKeyFile)
-      )) + "\n";
+    environment.etc."ssh/ssh_known_hosts".text = ''
+      ${(flip (concatMapStringsSep "\n") knownHosts
+        (h: assert h.hostNames != [];
+          concatStringsSep "," h.hostNames + " "
+          + (if h.publicKey != null then h.publicKey else readFile h.publicKeyFile)
+        )) + "\n"}
+      ${cfg.extraConfig}
```
Comment (LnL7) [inline modules/programs/ssh/default.nix]: I'm not sure how it's called in nixos, but this sounds a bit general given that it's only the known_hosts file.

Comment (luigy) [reply modules/programs/ssh/default.nix]: I actually noticed this was missing while trying to reuse an internal module from one of my nixos machines. Here is what nixos ssh module looks like https://github.com/NixOS/nixpkgs/blob/83fd9785f65b8fd4cbaa146270f90b2a0d74b5f3/nixos/modules/programs/ssh.nix#L217

Comment (LnL7) [reply modules/programs/ssh/default.nix]: Yeah, but that's `ssh_config` not `known_hosts`.

Comment (luigy) [reply modules/programs/ssh/default.nix]: Whoops wrong config

Comment (luigy) [reply modules/programs/ssh/default.nix]: moved to `ssh_config`

**Comment by Enzime** [comment]
Thanks for the contribution, however I'm going to close this PR as it's quite old and I think the approach would require some rework. On macOS, `/etc/ssh/ssh_config` exists by default which means this PR in its current form would clash with the existing `/etc/ssh/ssh_config`. It also reads configuration from `/etc/ssh/ssh_config.d/*` which means users can already change SSH settings by creating files like so:

```
environment.etc."ssh/ssh_config.d/100-my-configuration.conf".text = ''
  Host *
    ForwardAgent yes
'';
```

Please open a new PR, if you're still interested in reworking this approach, possibly to write extra configurations to a file in `/etc/ssh/ssh_config.d` with the configurations, or possibly introducing a new option to forcibly override `/etc/ssh/ssh_config`.

**Pull request closed without merging**